mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,382 @@
+from sklearn import metrics as sk_metrics
+import math
+import pandas as pd
+import numpy as np
+import time
+from functools import partial
+import logging
+
+"""
+[P0] Accuracy: Calculates how often predictions equal labels.
+[P0] BinaryCrossentropy: Computes the crossentropy metric between the labels and predictions.
+[P0] Hinge: Computes the hinge metric between y_true and y_pred.
+[P0] Sum: Computes the (weighted) sum of the given values.
+[P0] Mean: Computes the (weighted) mean of the given values.
+[P0] ExampleCount: Computes the total number of evaluation examples.
+[P0] MeanAbsoluteError: Computes the mean absolute error between the labels and predictions.
+[P0] MeanSquaredError: Computes the mean squared error between y_true and y_pred.
+[P0] RootMeanSquaredError: Computes root mean squared error metric between y_true and y_pred.
+
+[P0] TrueNegatives: Calculates the number of true negatives.
+[P0] TruePositives: Calculates the number of true positives.
+[P0] FalseNegatives: Calculates the number of false negatives.
+[P0] FalsePositives: Calculates the number of false positives.
+https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix
+
+[P0] Precision: Computes the precision of the predictions with respect to the labels.
+[P0] Recall: Computes the recall of the predictions with respect to the labels.
+[P0] AUC: Approximates the AUC (Area under the curve) of the ROC or PR curves.
+[P0] F1 Score: 2*precision*recall / (precision+recall)
+
+[P0] BinaryClassConfusionMatrix
+
+Plots
+[P0] Confusion matrix
+[P0] Interactive ROC curve with metrics (TP/TN/FP/FN/Acc/F1/AUC), binary classification
+[P0] Lift chart
+
+Global explainability
+[P0] Model built-in feature importance (supported models)
+[P0] SHAP explainers
+     [P0] Summary plot
+"""
+
+from PIL.Image import Image, open as open_image
+
+
+_logger = logging.getLogger(__name__)
+
+
+class ImageEvaluationArtifact(EvaluationArtifact):
+    def save(self, output_artifact_path):
+        self._content.save(output_artifact_path)
+
+    def _load_content_from_file(self, local_artifact_path):
+        self._content = open_image(local_artifact_path)
+        return self._content
+
+
+class CsvEvaluationArtifact(EvaluationArtifact):
+    def save(self, output_artifact_path):
+        self._content.to_csv(output_artifact_path, index=False)
+
+    def _load_content_from_file(self, local_artifact_path):
+        self._content = pd.read_csv(local_artifact_path)
+        return self._content
+
+
+_DEFAULT_SAMPLE_ROWS_FOR_SHAP = 2000
+
+
+class DefaultEvaluator(ModelEvaluator):
+    def can_evaluate(self, model_type, evaluator_config=None, **kwargs):
+        return model_type in ["classifier", "regressor"]
+
+    @staticmethod
+    def _gen_log_key(key, dataset_name, model):
+        if hasattr(model.metadata, 'model_uuid'):
+            return f'{key}_on_data_{dataset_name}_model_{model.metadata.model_uuid}'
```

Comment:
Not sure whether the key format is too long.

Reply:
Can we take the model UUID out of the key and instead add it as an attribute to the entry(s) in `mlflow.datasets`? e.g. `{"name":"...","hash":"...","path":"...","model":"..."}`? We want to make sure that users can easily compare these metrics across runs / models, so using the same metric name and excluding model UUID is important.